include/envoy/http/access_log.h

```diff
@@ -89,9 +90,11 @@ class RequestInfo {
   virtual std::chrono::milliseconds duration() const PURE;
 
   /**
-   * @return the failure reason for richer log experience.
+   * @return response flags to enrich access log with details around response code. Response flag
+   * complements response
+   * code and add details to it.
    */
-  virtual FailureReason failureReason() const PURE;
+  virtual bool isSetResponseFlag(ResponseFlag response_flag) const PURE;
```

Comment:
here i was following convention used in thrift for optional fields, getResponseFlag() should be ok as well.